prow/cmd/branchprotector/protect.go

```diff
@@ -186,144 +160,92 @@ func (p *Protector) ConfigureBranches() {
 
 // Protect branches specified in the presubmit and branch-protection config sections.
 func (p *Protector) Protect() {
-	bp := p.cfg.BranchProtection
-
-	// Scan the branch-protection configuration
-	for orgName, org := range bp.Orgs {
-		if err := p.UpdateOrg(orgName, org, bp.Protect, bp.Contexts, bp.Pushers); err != nil {
-			p.errors.add(err)
-		}
-	}
-
-	// Do not automatically protect tested repositories
-	if !bp.ProtectTested {
-		return
+	type orgRepo struct {
+		org, repo string
 	}
+	tmpRefs := map[orgRepo][]string{}
 
 	// Some repos with presubmits might not be listed in the branch-protection
-	for repo := range p.cfg.Presubmits {
-		if p.completedRepos[repo] == true {
-			continue
-		}
-		parts := strings.Split(repo, "/")
-		if len(parts) != 2 {
-			log.Fatalf("Bad repo: %s", repo)
-		}
-		orgName := parts[0]
-		repoName := parts[1]
-		repoReqs := repoRequirements(orgName, repoName, *p.cfg)
-		protect := len(repoReqs) > 0
-		if err := p.UpdateRepo(orgName, repoName, config.Repo{}, &protect, repoReqs, nil); err != nil {
-			p.errors.add(err)
-		}
-	}
-}
-
-// Update all repos in the org with the specified defaults
-func (p *Protector) UpdateOrg(orgName string, org config.Org, protect *bool, contexts, pushers []string) error {
-	if org.Protect != nil {
-		protect = org.Protect
-	}
-
-	oc := append([]string(nil), contexts...)
-	oc = append(oc, org.Contexts...)
-
-	op := append([]string(nil), pushers...)
-	op = append(op, org.Pushers...)
-
-	var repos []string
-	if protect != nil {
-		// Strongly opinionated org, configure every repo in the org.
-		rs, err := p.client.GetRepos(orgName, false)
-		if err != nil {
-			return fmt.Errorf("GetRepos(%s) failed: %v", orgName, err)
-		}
-		for _, r := range rs {
-			repos = append(repos, r.Name)
-		}
-	} else {
-		// Unopinionated org, just set explicitly defined repos
-		for r := range org.Repos {
-			repos = append(repos, r)
+	// Do not automatically protect tested repositories
```

Comment:
Please do not change default behavior in this PR

Reply:
I am not this is coming from https://sourcegraph.com/github.com/kubernetes/test-infra@f2213eb3e19d95930c192db44ac06dffc1f4caf9/-/blob/prow/cmd/branchprotector/protect.go#L198

Reply:
Note that yours tests were pretty good, so it was easy to make sure that my change kept the funcionality

Reply:
Given that I'm essentially in the middle of refactoring this whole file to support the rest of the github options, a secondary goal is to make it as easy as possible for me to understand what parts you need me to pull in